tuf/api/metadata.py

```diff
@@ -124,8 +124,7 @@ def from_dict(cls, metadata: Dict[str, Any]) -> "Metadata[T]":
             metadata: TUF metadata in dict representation.
 
         Raises:
-            KeyError: The metadata dict format is invalid.
-            ValueError: The metadata has an unrecognized signed._type field.
+            ValueError, KeyError: Invalid arguments.
```

Comment:
It is hard to believe we can't get TypeError here, considering we potentially call every other `from_dict()`?

Reply:
Yes, you are correct. Just calling `Key.from_dict()` will lead to `TypeError`.